pkg/scheduler/scheduler.go

```diff
@@ -98,6 +98,52 @@ type Scheduler struct {
 	nextStartNodeIndex int
 }
 
+// newScheduler creates a Scheduler object.
+func newScheduler(
+	cache internalcache.Cache,
+	client clientset.Interface,
+	nodeInfoSnapshot *internalcache.Snapshot,
+	percentageOfNodesToScore int32) *Scheduler {
+	sched := Scheduler{
+		Cache:                    cache,
+		client:                   client,
+		nodeInfoSnapshot:         nodeInfoSnapshot,
+		percentageOfNodesToScore: percentageOfNodesToScore,
+	}
+	return &sched
+}
+
+func (s *Scheduler) extenders(extenders []framework.Extender) *Scheduler {
+	s.Extenders = extenders
+	return s
+}
+
+func (s *Scheduler) nextPod(fc func() *framework.QueuedPodInfo) *Scheduler {
+	s.NextPod = fc
+	return s
+}
+
+func (s *Scheduler) stopEverything(ch <-chan struct{}) *Scheduler {
+	s.StopEverything = ch
+	return s
+}
+
+func (s *Scheduler) schedulingQueue(queue internalqueue.SchedulingQueue) *Scheduler {
+	s.SchedulingQueue = queue
+	return s
+}
+
+func (s *Scheduler) profiles(profiles profile.Map) *Scheduler {
+	s.Profiles = profiles
+	return s
+}
+
+func (s *Scheduler) do() *Scheduler {
+	s.SchedulePod = s.schedulePod
+	s.FailureHandler = s.handleSchedulingFailure
+	return s
+}
```

Comment:
`do()` doesn't read well IMO. Maybe `apply` or `defaults`.

Reply:
Change to apply()